recipes/dhcpcd/files/init

```diff
@@ -16,7 +16,12 @@ do_start() {
 }
 
 do_stop() {
-    start-stop-daemon -K --pidfile $PIDFILE --exec $DAEMON
+    # only try to stop the daemon if there is actually something
+    # running. Otherwise a restart before a start will fail
+    start-stop-daemon -K -o --pidfile $PIDFILE --exec $DAEMON
+    if [ "$?" != "0" ] ; then
+	start-stop-daemon -K --pidfile $PIDFILE --exec $DAEMON
+    fi
 }
```

Comment:
Yes, it would be absolutely ideal to have the action "stop" exit non-zero if an error occured or no daemon was running, while "restart" would ignore the latter case and proceed to the start part. However, the source doesn't suggest that's how it works:
```
	if (opt & CTX_STOP) {
		int i = do_stop();
		return (opt & OPT_OKNODO) ? 0 : (i <= 0);
	}
```